db/system_keyspace.cc

```diff
@@ -1953,6 +1953,36 @@ class token_ring_table : public streaming_virtual_table {
     }
 };
 
+// system.dual helper table
+class dual_table : public streaming_virtual_table {
```

Comment:
I'm afraid I'm missing the point.
With CALL syntax we can call functions - nice. With `dual` table we can call functions while pretending that we're SELECTing. So does `dual` solve any extra problem?

In this particular case `dual` table is a mock, it's even unused. It doesn't even accept any concrete function that would compute something.

What's the plan here?

Reply:
I'm not sure what were the reasons behind it, beside just a different syntax.
@kostja maybe you could clarify?

Reply:
It's incorrect in ANSI to CALL a  function. This was not in the original patch. CALL is for procedures, which return a result set, and FUNCTION returns just a single value. We could unify the api by returning a virtual table from both procedures and  functions, but I don't see this happen, besides, that would complicate using functions in scalar context.

Reply:
Functions can compose, while procedures do not (well, they do in stored procedures which we don't have). So there is room for both.